source/extensions/filters/http/aws_request_signing/aws_request_signing_filter.cc

```diff
@@ -20,20 +23,29 @@ Extensions::Common::Aws::Signer& FilterConfigImpl::signer() { return *signer_; }
 FilterStats& FilterConfigImpl::stats() { return stats_; }
 
 const std::string& FilterConfigImpl::hostRewrite() const { return host_rewrite_; }
+bool FilterConfigImpl::useUnsignedPayload() { return unsigned_payload_; }
 
 FilterStats Filter::generateStats(const std::string& prefix, Stats::Scope& scope) {
   const std::string final_prefix = prefix + "aws_request_signing.";
   return {ALL_AWS_REQUEST_SIGNING_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
 }
 
-Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
+Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers, bool end_stream) {
   const auto& host_rewrite = config_->hostRewrite();
+  bool unsigned_payload = config_->useUnsignedPayload();
+
   if (!host_rewrite.empty()) {
     headers.setHost(host_rewrite);
   }
 
+  if (!unsigned_payload && !end_stream) {
+    request_headers_ = &headers;
+    return Http::FilterHeadersStatus::StopIteration;
```

Comment:
Hm, I inferred this as necessary because it's what the buffer filter and the aws lambda filter do. I supposed it was necessary to preclude the headers from being sent upstream before the stream ends.

I don't mind adding a comment, but the best I can honestly write with my current understanding is "stop iteration is what other filters do when the request is to be buffered."

Did you have something more concrete in mind?

Reply:
My experience with filter flows is limited, but I think you are right: StopIteration is needed to buffer the request.

I don't have strong opinion if a comment is needed. Let's leave it as it is.